File: Graphs/Implementation.c
#include <stdio.h>
#include <stdlib.h>

struct Node
{
    int data;
    struct Node *next;
};

void create(struct Node *ad[],int n)
{
    struct Node *newNode;
    int i,j,k,val;

    for(i=0;i<n;i++)
    {
        struct Node *last=NULL;
        printf("\nEnter the number of neighbours of %d",i+1);
        scanf("%d",&k);

        for(j=0;j<k;j++)
        {
            printf("Enter the value of %d neighbours of %d:",j+1,i+1);
            scanf("%d",&val);
            newNode = (struct Node*)malloc(sizeof(struct Node*));
            newNode->data=val;
            newNode->next=NULL;
            if(ad[i]==NULL)
                ad[i]=newNode;
            else
                last->next=newNode;
            
            last=newNode;
            
        }
    }
}

void printGraph(struct Node *ad[],int n)
{
   struct Node *ptr=NULL;
   int i,j;
   for(i=0;i<n;i++)
   {
       ptr=ad[i];
       printf("The neighbours of %d are",i+1);
       while(ptr!=NULL)
       {
           printf("%d\t",ptr->data);
           ptr=ptr->next;
       }

       printf("\n");
   }
}


void main()
{
    int i,j,k,nodes;
    printf("Enter the number of nodes: ");
    scanf("%d",&nodes);

    struct Node *ad[nodes];
    for(i=0;i<nodes;i++)
    {
        ad[i]=NULL;
    }

    create(ad,nodes);
    printGraph(ad,nodes);
}

File: Graphs/Implementation.java
import java.util.*;

public class Implementation 
{
    static Scanner scan = new Scanner(System.in);
   

    static void readGraph(TreeMap<Integer,ArrayList<Integer>> graph,int n)
    {
          int i,j,k,val;
          for(i=0;i<n;i++)
          {
              System.out.println("Enter the number of neighbours of :"+(i+1));
              k=scan.nextInt();

              ArrayList<Integer> list = new ArrayList<>();

              for(j=0;j<k;j++)
              {
                  System.out.println("Enter the value of "+(j+1)+" neighbours of:"+(i+1));
                  val = scan.nextInt();
                  list.add(val);
                  graph.put(i,list);
              }
          }
    }

    static void printGraph(TreeMap<Integer,ArrayList<Integer>> graph,int n)
    {
                 for(Map.Entry<Integer,ArrayList<Integer>> entry: graph.entrySet())
                 {
                      int key = entry.getKey()+1; 
                      ArrayList<Integer> al = entry.getValue();
                   
                      System.out.print("The neighbours of "+key+" are:");
                    for(Integer i:al)
                    {
                        System.out.print(i+" ");
                    }

                    System.out.println();
                    System.out.println();

                 }
    }



    public static void main(String[] args) 
    {
        int nodes;
        System.out.println("Enter the number of nodes:");
        nodes = scan.nextInt();
        System.out.println("Number of nodes:"+nodes);

        TreeMap<Integer,ArrayList<Integer>> graph = new  TreeMap<>();

        readGraph(graph,nodes);
        printGraph(graph,nodes);

        
    }
    
}
